shapely/geometry/base.py

```diff
@@ -64,6 +64,14 @@ def dump_coords(geom):
         raise GeometryTypeError("Unhandled geometry type: " + repr(geom.geom_type))
 
 
+def _maybe_unpack(result, typ):
+    if result.ndim == 0:
+        return typ(result)
```

Comment:
Did you try `result.item()`? That would prevent the separate type administration.

Reply:
Good point, that's indeed simpler!